### PR TITLE
Optical Code Improvements

### PR DESCRIPTION
**Pull request by FaserF**
# Proposed Changes
This PR will do some optical code improvements based on recommendations by Microsoft about writing Powershell code.
This commit also prepares to trigger the Megalinter Code analysis, which unfortunatly currently gets skipped (I dont know why, but a open issue about this is here: https://github.com/oxsecurity/megalinter/issues/2497 ). Otherwise Megalinter would do those improvements automatically with a auto generated PR.

This PR is a DRAFT, until the Megalinter trigger gets started. The reason why I have started a draft, is that you and other PR submitter can see that here is a WIP, that could potentially make merging other stuff more complicated.

## Related Issues

> ([Github link][autolink-references] to related issues or pull requests)

[autolink-references]: https://help.github.com/articles/autolinked-references-and-urls/

### CODE REVIEW

**Comment by FaserF** [comment]
Megalinter now works as expected and the PR could be merged. 
Note: Write-Log function has been renamed to Write-ToLog, as PS has its own integrated function which would be written the same way.
All PS1 files have been touched to improve the readability. Most likely Megalinter is going to create another automated PR with more changes.

File: Winget-AutoUpdate/User-Run.ps1
Comment (KnifMelti) [inline Winget-AutoUpdate/User-Run.ps1:1]: You can't comment out the common variables in this file, they're used in the included Start-NotifTask.ps1!

File: Winget-AutoUpdate/User-Run.ps1 [hunk 1]
```diff
@@ -44,9 +44,9 @@ Get-NotifLocale
 #Set common variables
 $OnClickAction = "$WorkingDir\logs\updates.log"
 $Button1Text = $NotifLocale.local.outputs.output[11].message
-$Title = "Winget-AutoUpdate (WAU)"
-$Balise = "Winget-AutoUpdate (WAU)"
-$UserRun = $True
```
Comment (Romanitho) [inline Winget-AutoUpdate/User-Run.ps1:49]: @KnifMelti, I agree

**Review by Romanitho — COMMENTED** [review]
@FaserF we need to keep these variables uncommented

File: Winget-AutoUpdate/User-Run.ps1 (hunk 1 above)
Comment (KnifMelti) [reply Winget-AutoUpdate/User-Run.ps1:49]: If not we must in **Start-NotifTask.ps1** introduce 
```
$caller = Get-ChildItem $MyInvocation.PSCommandPath | Select-Object -Expand Name
    if ($caller -eq "User-Run.ps1") {...}
```

File: Winget-AutoUpdate/User-Run.ps1
Comment (FaserF) [reply Winget-AutoUpdate/User-Run.ps1:1]: thanks for the hint! Didnt knew that.

**Comment by Romanitho** [comment]
Thanks

**Pull request merged**